doc/src/Intro.rst

```diff
@@ -76,6 +90,12 @@ For dependencies on other boost libraries, they can be listed with the ``DEPENDS
             config
     )
 
+.. This is interesting. I like the reduction in boilerplate, but I question the
+   choice to go in such a different direction from the standard CMake way of
+   doing things. It strikes me as a possibly leaky abstraction where folks will
+   have a mix of code which uses both the CMake way and the BCM way; maintainers
+   will have to be familiar with both.
```

Comment:
The `bcm_boost_package` is setup this way so it can handle both superproject builds and modular builds. Ideally, we could remove `DEPENDS` and have it work like `bcm_package`, like this:

```cmake
bcm_find_package(boost_assert)
bcm_find_package(boost_config)

bcm_boost_package(boost_core
    VERSION 1.61.0
)

target_link_libraries(boost_core boost::assert boost::config)
```

We will just need to hijack `find_package` for superproject builds. 

Of course, I like the terse version more. I don't think its a problem mixing bcm functions with cmake functions. Its designed to do that. Also, I think it helps authors who are not familiar with cmake. 

Finally, there is no tool to ensure the correctness of the above snippet. An author may do:

```cmake
bcm_find_package(boost_assert)
bcm_find_package(boost_config)

bcm_boost_package(boost_core
    VERSION 1.61.0
)

target_link_libraries(boost_core boost::config)
```

Which is a mistake, but there is no way to detect this mistake.

Of course, for external dependencies, it needs to be done like above:

```cmake
bcm_find_package(ZLib)

bcm_boost_package(boost_core
    VERSION 1.61.0
    DEPENDS
        assert
        config
)

target_link_libraries(boost_core Zlib::ZLib)
``` 

However, boost has very little external dependencies.